bots/push-rewrite

```diff
@@ -0,0 +1,66 @@
+#! /usr/bin/python3
+
+# push-rewrite -- Force push to a PR after rewriting history, with benefits.
+#
+# This tool force pushes your local changes to origin but only if that
+# doesn't change any files.  If that is the case, the tool will also
+# copy the test results over.
+#
+# The idea is that you use this tool after squashing fixups in a pull
+# request or after other similar last minute rewriting activity before
+# merging a pull request.  Then you can merge the PR using the GitHub
+# UI and get a nice "merged" label for it, and the tests will still be
+# green.
+
+import subprocess
+import argparse
+import sys
+
+sys.dont_write_bytecode = True
```

Comment:
Can we drop this? Python doesn't create .pyc files for executables anyway.

Reply:
Yes, removed.  I cargo-culted that from somewhere...